include/nbl/builtin/hlsl/sampling/quotient_and_pdf.hlsl

```diff
@@ -0,0 +1,49 @@
+// Copyright (C) 2018-2025 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_BUILTIN_HLSL_SAMPLING_QUOTIENT_AND_PDF_INCLUDED_
+#define _NBL_BUILTIN_HLSL_SAMPLING_QUOTIENT_AND_PDF_INCLUDED_
+
+#include "nbl/builtin/hlsl/concepts/vector.hlsl"
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace sampling
+{
+
+template<typename T, typename F>
+NBL_BOOL_CONCEPT spectral_of = concepts::Vectorial<T> || concepts::Scalar<T> || concepts::Scalar<F>;
+
+// finally fixed the semantic F-up, value/pdf = quotient not remainder
+template<typename Q, typename P NBL_PRIMARY_REQUIRES(spectral_of<Q,P> && is_floating_point_v<P>)
```

Comment:
`Q` should also be floating point, I don't see a useful world in which it could be an integer or boolean